src/Systems.jl

```diff
@@ -4,34 +4,44 @@ import Random
 Defines a collection of spins, as well as the Hamiltonian they interact under.
  This is the main type to interface with most of the package.
 """
-struct SpinSystem{N}
+mutable struct SpinSystem{N}   ## Note: Changed to mutable so old integrator would work
     lattice     :: Lattice                          # Definition of underlying lattice
     hamiltonian :: HamiltonianCPU                   # Contains all interactions present
     _dipoles    :: Array{Vec3, 4}                   # Holds dipole moments: Axes are [Basis, CellA, CellB, CellC]
     _coherents  :: Array{SVector{N, ComplexF64}, 4} # Coherent states
     site_infos  :: Vector{SiteInfo}                 # Characterization of each basis site
+    S           :: NTuple{3, Matrix{ComplexF64}}
 end
 
 @inline Base.size(sys::SpinSystem) = size(sys._coherents)
 @inline nbasis(sys::SpinSystem) = nbasis(sys.lattice)
 @inline eachcellindex(sys::SpinSystem) = eachcellindex(sys.lattice)
 
+@inline _get_coherent_from_dipole(dip::Vec3, ::Val{0}) :: CVec{0} = CVec{0}(zeros(0))
 @inline function _get_coherent_from_dipole(dip::Vec3, ::Val{N}) :: CVec{N} where {N} 
-    # TODO: Produce a coherent state of SU(N) agreeing with the given dipole moment.
-    return CVec{N}(zeros(N))
+    S = gen_spin_ops(N) 
+    λs, vs = eigen(dip⋅S)
+    return CVec{N}(vs[:, argmax(real.(λs))])
 end
 
+# Construct an analogous CoherentView?
 struct DipoleView{N} <: AbstractArray{Vec3, 4}
     _dipoles    :: Array{Vec3, 4}
     _coherents  :: Array{SVector{N, ComplexF64}, 4}
 end
 
 Base.IndexStyle(::Type{DipoleView}) = IndexLinear()
 Base.size(dv::DipoleView) = size(dv._dipoles)
-Base.getindex(dv::DipoleView, i) = getindex(dv._dipoles, i)
```

Comment:
I might not be following, why remove the generic `getindex` and only support the `Int` and `Vararg{Int, M}` cases?

Reply:
(Same with `setindex!`)

Reply:
I think you are right. A few tests started to fail and added those methods to fix them. But, looking at it now, it doesn't seem like it should be necessary.  I'll review and revert back to the generic case if it's appropriate.

Reply:
Think I was actually getting errors because Vec3 was ambiguous in the environment I was testing in, but the error message didn't make this clear.  (Makie also defines a Vec3.) Reveting to the generic implementation as suggested.